api/types.proto

```diff
@@ -30,12 +30,31 @@ message Annotations {
 	repeated IndexEntry indices = 4 [(gogoproto.nullable) = false];
 }
 
+message Set {
+	message Void {}
+	map<string, Void> val = 1;
+}
+
+message Integer {
+	uint64 val = 1;
+}
+
+message GenericResource {
+	oneof resource {
+		Integer integer = 1;
```

Comment:
I think it's possible to put an `uint64` directly inside the `oneof`. I see an example at https://developers.google.com/protocol-buffers/docs/proto#oneof with a `string` inside a `oneof`. I suspect the map could go directly in there as well. If there aren't technical issues with avoiding the `Set` and `Integer` types, it might make things a bit simpler. There would be fewer levels of indirection involved.